### DISCUSSION TITLE
Can't use custom timetable

### DISCUSSION THREAD

**Discussion by taylorfinnell** [apache/airflow#29798] [General]
### Apache Airflow version

2.5.1

### What happened

When implementing a custom timetable which is based on the included `workday` timetable I get runtime errors.

`ValueError: Not a valid timetable`

```
/opt/app-root/lib64/python3.8/site-packages/airflow/utils/session.py:75: in wrapper
    return func(*args, session=session, **kwargs)
tests/helpers.py:34: in run_dag
    dag.test(
/opt/app-root/lib64/python3.8/site-packages/airflow/utils/session.py:75: in wrapper
    return func(*args, session=session, **kwargs)
/opt/app-root/lib64/python3.8/site-packages/airflow/models/dag.py:2518: in test
    dr: DagRun = _get_or_create_dagrun(
/opt/app-root/lib64/python3.8/site-packages/airflow/models/dag.py:3676: in _get_or_create_dagrun
    dr = dag.create_dagrun(
/opt/app-root/lib64/python3.8/site-packages/airflow/utils/session.py:72: in wrapper
    return func(*args, **kwargs)
/opt/app-root/lib64/python3.8/site-packages/airflow/models/dag.py:2587: in create_dagrun
    data_interval = self.infer_automated_data_interval(logical_date)
```

My understanding is that `infer_automated_data_interval` should never be used or called when a custom timetable is being used.

My custom timetable implements only the following methods:

`infer_manual_data_interval` and `next_dagrun_info`.

### What you think should happen instead

Custom timetable should allow scheduling

### How to reproduce

I believe this issue will happen when using the built in `workday` timetable in the examples

### Operating System

Centos 7

### Deployment

Other

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template!

**Comment by taylorfinnell** [comment]
Ah, this might be because I was using `DAG#test`, seems to work as a BackfillJob

**Reply by potiuk** [reply]
Yes could be but maybe @uranusjr can confirm

**Reply by taylorfinnell** [reply]
I can confirm that if I monkey patch `_get_or_create_dagrun` to  include a `data_interval` param when it calls `dag.create_dagrun` it works just fine

**Reply by uranusjr** [reply]
Sounds like a bug, and the fix (adding an explicit `data_interval` param) should be correct. Open a PR please!
